### PR TITLE
Support RPC v0.2.1

### PR DESCRIPTION
**Pull request by drknzz**
## Adapt to [RPC v0.2.1](https://github.com/starkware-libs/starknet-specs/releases/tag/v0.2.1)

Significantly smaller PR than for previous versions, introducing changes in RPC v0.2.0 - v0.2.1.

## Checklist:

- [x] Applied formatting - `./scripts/format.sh`
- [x] No linter errors - `./scripts/lint.sh`
- [x] Performed code self-review
- [x] Rebased to the last commit of the target branch (or merged it into my branch)
- [x] Documented the changes
- [x] Updated the tests
- [x] All tests are passing

### CODE REVIEW

File: starknet_devnet/blueprints/rpc/structures/payloads.py [hunk 1]
```diff
@@ -314,6 +336,28 @@ def rpc_deploy_transaction(transaction: DeploySpecificInfo) -> RpcDeployTransact
     return txn
 
 
+def rpc_deploy_account_transaction(
```
Comment (FabijanC) [inline starknet_devnet/blueprints/rpc/structures/payloads.py:339]: Where is this used?

**Review by FabijanC — CHANGES_REQUESTED** [review]

File: starknet_devnet/blueprints/rpc/structures/payloads.py (hunk 1 above)
Comment (drknzz) [reply starknet_devnet/blueprints/rpc/structures/payloads.py:340]: Nowhere actually, good catch.

File: starknet_devnet/blueprints/rpc/transactions.py [hunk 2]
```diff
@@ -143,6 +146,33 @@ async def add_deploy_transaction(deploy_transaction: RpcBroadcastedDeployTxn) ->
...
+    """
+    Submit a new deploy account transaction
+    """
+    deploy_account_tx = make_deploy_account(deploy_account_transaction)
+
+    contract_address, transaction_hash = await state.starknet_wrapper.deploy_account(
+        external_tx=deploy_account_tx
+    )
+
+    status_response = state.starknet_wrapper.transactions.get_transaction_status(
```
[the remark this answers is not in the document]
Comment (drknzz) [reply starknet_devnet/blueprints/rpc/transactions.py:161]: It's a way to check if `CLASS_HASH_NOT_FOUND` RPC error should be thrown. In [specification](https://github.com/starkware-libs/starknet-specs/blob/v0.2.1/api/starknet_write_api.json#L138), only deploy_account transactions throw `CLASS_HASH_NOT_FOUND`.

File: test/rpc/test_rpc_transactions.py [hunk 3]
```diff
@@ -7,6 +7,10 @@
 from typing import List
 
 from test.account import declare, _get_execute_args, get_nonce, _get_signature
+from test.rpc.conftest import (
+    _prepare_deploy_account_tx,
```
Comment (FabijanC) [inline test/rpc/test_rpc_transactions.py:11]: Isn't the convention for single-underscore-prepended functions to only be used internally (like within a class or a file)? If so, could you rename these?

File: test/rpc/test_rpc_transactions.py [hunk 4]
```diff
@@ -607,3 +689,115 @@ def test_add_deploy_transaction(deploy_content, version):
...
+
+    ex = rpc_call(
+        "starknet_addDeployAccountTransaction",
+        params={"deploy_account_transaction": rpc_deploy_account_tx},
+    )
+    assert ex["error"] == {"code": 28, "message": "Class hash not found"}
+
+
+@pytest.mark.usefixtures("run_devnet_in_background")
+@pytest.mark.parametrize(
```
Comment (FabijanC) [inline test/rpc/test_rpc_transactions.py:729]: Why did you parametrize this test when there's only one case? The next test case as well.

**Review by FabijanC — CHANGES_REQUESTED** [review]
Just to make sure I understood the new changes, you added two new tests: `test_get_transaction_by_hash_deploy_account` and `test_get_deploy_account_transaction_receipt`?

File: test/rpc/test_rpc_transactions.py (hunk 4 above)
Comment (drknzz) [reply test/rpc/test_rpc_transactions.py:729]: I thought it would make the test clearer. I can move them back inside the tests though.

**Review by FabijanC — APPROVED** [review]

**Pull request merged**